### PR TITLE
Improves reliability of the db restart test for content app

### PR DESCRIPTION
**Pull request by dkliban**
[noissue]

### CODE REVIEW

**Review by bmbouter — APPROVED** [review]
Thanks

**Comment by mdellweg** [comment]
Would it be possible to query the status endpoint to check for the service health? I'd prefer something more specific than a generic failure in the setup method.

**Comment by dkliban** [comment]
> Would it be possible to query the status endpoint to check for the service health? I'd prefer something more specific than a generic failure in the setup method.

We could check the status API, but we would still need to check for an ApiException. The status API is going to return 500 error when the db is not ready. Though I still think the test would be better that way. I'll update it.

**Comment by bmbouter** [comment]
I like this version even more. Double LGTM!

File: pulpcore/tests/functional/api/using_plugin/utils.py [hunk 1]
```diff
@@ -61,6 +61,12 @@ def populate_pulp(cfg, url=None):
     return client.get(FILE_CONTENT_PATH)
 
 
+def gen_pulpcore_client():
```
Comment (mdellweg) [inline pulpcore/tests/functional/api/using_plugin/utils.py:64]: I will love to see these turned into pytest fixtures.

**Review by mdellweg — APPROVED** [review]

File: pulpcore/tests/functional/api/using_plugin/utils.py (hunk 1 above)
Comment (bmbouter) [reply pulpcore/tests/functional/api/using_plugin/utils.py:64]: Me too. I think my PR work is on the way to enabling that.

**Pull request merged**